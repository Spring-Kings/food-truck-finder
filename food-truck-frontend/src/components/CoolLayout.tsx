import SiteHeader from './SiteHeader'
import SiteFooter from './SiteFooter'

import styles from './layout.module.css'

import Head from 'next/head'
import {ReactNode} from 'react'
import {Container} from '@material-ui/core'

export const siteTitle = 'Stacked Trucks'

type Args = {
    children: ReactNode
}

function coolLayout(args: Args) {
    return (
        <>
            <Head>
                <link rel="icon" href="/favicon.ico"/>
                <meta name="description" content="Oi mate description goes here"/>
            </Head>
            <div className={styles.outside}>
                <SiteHeader/>
                <Container maxWidth="md" className={styles.content}>
                    <main>{args.children}</main>
                </Container>
                <SiteFooter/>
            </div>
        </>
    );
}

export default coolLayout;