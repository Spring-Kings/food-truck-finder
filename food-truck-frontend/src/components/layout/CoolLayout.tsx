import AppMenuBarComponent from './AppMenuBar'
import SiteFooter from '../SiteFooter'

import Head from 'next/head'
import React, {ReactNode} from 'react'
import {Box, Container, createStyles, Grid, Paper, Theme} from '@material-ui/core'
import {makeStyles} from "@material-ui/core/styles";

export const siteTitle = 'Stacked Trucks'

export type Args = {
  children: ReactNode
}

const useLayoutStyles = makeStyles((theme: Theme) =>
  createStyles({
    paper: {
      minHeight: '100vh',
      flexGrow: 1,
    },
  }),
);

function coolLayout(args: Args) {
  const classes = useLayoutStyles();
  return (
    <>
      <Head>
        <link rel="icon" href="/favicon.ico"/>
        <meta name="description" content="Oi mate description goes here"/>
      </Head>
      <Grid direction="row">
        <Grid item>
          <Box mb={4}>
            <AppMenuBarComponent/>
          </Box>
        </Grid>
        <Grid item>
          <Container maxWidth="xl">
            <Paper className={classes.paper}>
              <Box p={4}>
                <Container>
                  <main>{args.children}</main>
                </Container>
              </Box>
            </Paper>
          </Container>
        </Grid>
        <Grid item>
          <Box p={2}>
            <SiteFooter/>
          </Box>
        </Grid>
      </Grid>
    </>
  );
}

export default coolLayout;