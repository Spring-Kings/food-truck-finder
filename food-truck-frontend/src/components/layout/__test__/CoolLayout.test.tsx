import React from "react";

import { shallow } from "enzyme";
import CoolLayout from "../CoolLayout"
import { Provider } from "react-redux";
import mockStore from "../../../util/test_util/mock_store";

describe("Snapshot test app menu bar", () => {
    test("Renders owner correctly", () => {
        const shallowRender = shallow(
            <Provider store={mockStore()}>
                <CoolLayout children={<div/>} />
            </Provider>
        );
        expect(shallowRender.html()).toMatchSnapshot();
    });

    test("Renders non-owner correctly", () => {
        const shallowRender = shallow(
            <Provider store={mockStore()}>
                <CoolLayout children={<div/>} />
            </Provider>
        );
        expect(shallowRender.html()).toMatchSnapshot();
    });
})
