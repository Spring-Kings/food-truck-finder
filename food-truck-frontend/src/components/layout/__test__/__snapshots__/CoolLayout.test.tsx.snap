// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test app menu bar Renders non-owner correctly 1`] = `"<div class=\\"MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center\\"><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12\\"><div class=\\"MuiBox-root MuiBox-root-7\\"><header class=\\"MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-root-3 mui-fixed MuiPaper-elevation4\\"><div class=\\"MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-align-content-xs-space-between\\"><div class=\\"MuiGrid-root MuiGrid-item\\"><div class=\\"MuiAvatar-root MuiAvatar-rounded\\"><img alt=\\"logo\\" src=\\"/logo.png\\" class=\\"MuiAvatar-img\\"/></div></div><div class=\\"MuiGrid-root MuiGrid-item\\"><h6 class=\\"MuiTypography-root MuiTypography-h6\\">Stacked Trucks</h6></div><div class=\\"MuiGrid-root MuiGrid-item\\"><li class=\\"MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button\\" tabindex=\\"-1\\" role=\\"menuitem\\" aria-disabled=\\"false\\">Home</li></div><div class=\\"MuiGrid-root MuiGrid-item\\"><li class=\\"MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button\\" tabindex=\\"-1\\" role=\\"menuitem\\" aria-disabled=\\"false\\">Search Trucks</li></div><div class=\\"MuiGrid-root MuiGrid-item\\"><li class=\\"MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button\\" tabindex=\\"-1\\" role=\\"menuitem\\" aria-disabled=\\"false\\">Nearby Trucks</li></div><div class=\\"MuiGrid-root MuiGrid-item\\"><li class=\\"MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button\\" tabindex=\\"-1\\" role=\\"menuitem\\" aria-disabled=\\"false\\">Recommended Trucks</li></div><div class=\\"MuiGrid-root MuiGrid-item\\"><li class=\\"MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button\\" tabindex=\\"-1\\" role=\\"menuitem\\" aria-disabled=\\"false\\">Waluigi&#x27;s Dashboard</li></div><div class=\\"MuiBox-root MuiBox-root-8\\"><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z\\"></path></svg></span></button><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 22c1.1 0 2-.9 2-2h-4c0 1.1.89 2 2 2zm6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2z\\"></path></svg></span></button><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeSmall\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69zM12 18c-3.31 0-6-2.69-6-6s2.69-6 6-6 6 2.69 6 6-2.69 6-6 6zm0-10c-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4z\\"></path></svg></span></button><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit\\" tabindex=\\"0\\" type=\\"button\\" aria-controls=\\"bar-menu\\" aria-haspopup=\\"true\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z\\"></path></svg></span></button></div></div></div></header></div></div><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12\\"><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-md-10 MuiGrid-grid-lg-10 MuiGrid-grid-xl-10\\"><div class=\\"MuiPaper-root makeStyles-paper-1 MuiPaper-elevation0 MuiPaper-rounded\\"><div class=\\"MuiBox-root MuiBox-root-9\\"><div class=\\"MuiContainer-root MuiContainer-maxWidthLg\\"><main><div></div></main></div></div></div></div></div><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12\\"><div class=\\"MuiBox-root MuiBox-root-10\\"><div class=\\"MuiGrid-root\\">© Spring Kings 2020</div></div></div></div>"`;

exports[`Snapshot test app menu bar Renders owner correctly 1`] = `"<div class=\\"MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center\\"><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12\\"><div class=\\"MuiBox-root MuiBox-root-2\\"><header class=\\"MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-root-3 mui-fixed MuiPaper-elevation4\\"><div class=\\"MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-align-content-xs-space-between\\"><div class=\\"MuiGrid-root MuiGrid-item\\"><div class=\\"MuiAvatar-root MuiAvatar-rounded\\"><img alt=\\"logo\\" src=\\"/logo.png\\" class=\\"MuiAvatar-img\\"/></div></div><div class=\\"MuiGrid-root MuiGrid-item\\"><h6 class=\\"MuiTypography-root MuiTypography-h6\\">Stacked Trucks</h6></div><div class=\\"MuiGrid-root MuiGrid-item\\"><li class=\\"MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button\\" tabindex=\\"-1\\" role=\\"menuitem\\" aria-disabled=\\"false\\">Home</li></div><div class=\\"MuiGrid-root MuiGrid-item\\"><li class=\\"MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button\\" tabindex=\\"-1\\" role=\\"menuitem\\" aria-disabled=\\"false\\">Search Trucks</li></div><div class=\\"MuiGrid-root MuiGrid-item\\"><li class=\\"MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button\\" tabindex=\\"-1\\" role=\\"menuitem\\" aria-disabled=\\"false\\">Nearby Trucks</li></div><div class=\\"MuiGrid-root MuiGrid-item\\"><li class=\\"MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button\\" tabindex=\\"-1\\" role=\\"menuitem\\" aria-disabled=\\"false\\">Recommended Trucks</li></div><div class=\\"MuiGrid-root MuiGrid-item\\"><li class=\\"MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button\\" tabindex=\\"-1\\" role=\\"menuitem\\" aria-disabled=\\"false\\">Waluigi&#x27;s Dashboard</li></div><div class=\\"MuiBox-root MuiBox-root-4\\"><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z\\"></path></svg></span></button><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 22c1.1 0 2-.9 2-2h-4c0 1.1.89 2 2 2zm6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2z\\"></path></svg></span></button><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeSmall\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69zM12 18c-3.31 0-6-2.69-6-6s2.69-6 6-6 6 2.69 6 6-2.69 6-6 6zm0-10c-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4z\\"></path></svg></span></button><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit\\" tabindex=\\"0\\" type=\\"button\\" aria-controls=\\"bar-menu\\" aria-haspopup=\\"true\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z\\"></path></svg></span></button></div></div></div></header></div></div><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12\\"><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-md-10 MuiGrid-grid-lg-10 MuiGrid-grid-xl-10\\"><div class=\\"MuiPaper-root makeStyles-paper-1 MuiPaper-elevation0 MuiPaper-rounded\\"><div class=\\"MuiBox-root MuiBox-root-5\\"><div class=\\"MuiContainer-root MuiContainer-maxWidthLg\\"><main><div></div></main></div></div></div></div></div><div class=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12\\"><div class=\\"MuiBox-root MuiBox-root-6\\"><div class=\\"MuiGrid-root\\">© Spring Kings 2020</div></div></div></div>"`;
